migrations/v7.js

```diff
@@ -0,0 +1,144 @@
+  });
+  mutateContent('adapt-contrib-textInput - add ariaQuestion', async () => {
+    components.forEach(component => {
+      component.ariaQuestion = '';
+    });
+    return true;
+  });
+  checkContent('adapt-contrib-textInput - check ariaQuestion added', async () => {
+    const isValid = components.every(component => _.has(component, 'ariaQuestion'));
+    if (!isValid) throw new Error('ariaQuestion not added');
+    return true;
+  });
+  updatePlugin('adapt-contrib-textInput - update to v7.2.2', { name: 'adapt-contrib-textInput', version: '7.2.2', framework: '>=5.19.1' });
+});
+
+/**
+ * correct v1 schemas - legacy schema correct in v2.0.0 and updated in v3.0.0
+ */
+describe('adapt-contrib-textInput - v4.2.0 to v7.2.6', async () => {
+  let course;
+  const incorrectGlobalsTextInputPath = '_globals._textInput';
+  const ariaRegionPath = '_globals._components._textInput.ariaRegion';
+  whereFromPlugin('adapt-contrib-textInput - from v4.2.0 to v7.2.6', { name: 'adapt-contrib-textInput', version: '<7.2.6' });
+  mutateContent('adapt-contrib-textInput - update _globals ariaRegion', async () => {
+    course = getCourse();
+    if (!_.has(course, ariaRegionPath)) _.set(course, ariaRegionPath, 'Text input. Type your answer and then submit.');
+    _.unset(course, incorrectGlobalsTextInputPath);
+    return true;
+  });
+  checkContent('adapt-contrib-textInput - check _globals ariaRegion updated', async () => {
+    const isValid = !_.has(course, incorrectGlobalsTextInputPath) && _.has(course, ariaRegionPath);
+    if (!isValid) throw new Error(`${ariaRegionPath} default not updated`);
+    return true;
+  });
+  updatePlugin('adapt-contrib-textInput - update to v7.2.6', { name: 'adapt-contrib-text', version: '7.2.6', framework: '>=5.19.1' });
+});
+
+describe('adapt-contrib-textInput - v2.0.0 to v7.2.7', async () => {
+  let components;
+  const feedbackAltTitlePath = '_feedback.altTitle';
+  whereFromPlugin('adapt-contrib-textInput - from v2.0.0 to v7.2.7', { name: 'adapt-contrib-textInput', version: '<7.2.7' });
+  whereContent('adapt-contrib-textInput - where missing altTitle', async () => {
+    components = getComponents('textinput').filter(component => !_.has(component, feedbackAltTitlePath));
+    return Boolean(components.length);
+  });
+  mutateContent('adapt-contrib-textInput - add altTitle', async () => {
+    components.forEach(component => {
+      _.set(component, feedbackAltTitlePath, '');
+    });
+    return true;
+  });
+  checkContent('adapt-contrib-textInput - check altTitle added', async () => {
+    const isValid = components.every(component => _.has(component, feedbackAltTitlePath));
+    if (!isValid) throw new Error(`${feedbackAltTitlePath} not added`);
+    return true;
+  });
+  updatePlugin('adapt-contrib-textInput - update to v7.2.7', { name: 'adapt-contrib-textInput', version: '7.2.7', framework: '>=5.19.1' });
+});
+
+describe('adapt-contrib-textInput - v2.0.0 to v7.3.0', async () => {
```

Comment:
Minor point, but all the describ/whereFromPlugins say from v2.0.0, could this be changed to the previous version from the one we're updating to.

Reply:
I know the describe versions don't actually have any impact, but I have only been including specific base versions if the migration was related to something being changed from a previous migration task, such as:
https://github.com/adaptlearning/adapt-contrib-textInput/blob/48d4247102e82b4eff849800c0c9f32d2d3bc5e8/migrations/v7.js#L62

Where new attributes have been added, I would have thought these would go all the way back to the first migration supported release rather than just the previous one?